java-frontend/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -108,86 +92,48 @@ public V get(K key) {
   }
 
   @Override
-  public void forEach(PSet.Consumer<K> action) {
+  public void forEach(Consumer<K> action) {
     forEach(this, action);
   }
 
   @SuppressWarnings("unchecked")
-  private void forEach(AVLTree t, PSet.Consumer<K> action) {
+  private void forEach(AVLTree t, Consumer<K> action) {
     if (t.isEmpty()) {
       return;
     }
     forEach(t.left(), action);
-    action.accept((K) t.key());
     forEach(t.right(), action);
+    while (t != null) {
+      action.accept((K) t.key());
+      t = t.nextInBucket();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "t". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1226)